p/riscv_test.h

```diff
@@ -149,6 +157,11 @@ reset_vector:                                                           \
         la t0, trap_vector;                                             \
         csrw mtvec, t0;                                                 \
         CHECK_XLEN;                                                     \
+        /* check if S-mode is implemented                            */ \
+        csrr t0, misa;                                                  \
+        srli t0, t0, 18;                                                \
+        andi t0, t0, 1;                                                 \
+        beqz t0, 1f;                                                    \
```

Comment:
Tests shouldn't define stvec_handler if they aren't meant to run in S-mode.  It's better not to have this new dynamic check, because this sort of misconfiguration should fail noisily.